pinot-segment-local/src/main/java/org/apache/pinot/segment/local/recordtransformer/DataTypeTransformer.java

```diff
@@ -40,14 +47,24 @@
  */
 @SuppressWarnings("rawtypes")
 public class DataTypeTransformer implements RecordTransformer {
+  private static final Logger LOGGER = LoggerFactory.getLogger(DataTypeTransformer.class);
+
   private final Map<String, PinotDataType> _dataTypes = new HashMap<>();
+  private boolean _useDefaultValueOnError;
+  private DateTimeFieldSpec _timeColumnSpec;
```

Comment:
We may store the `_timeColumnName` and `_timeFormatSpec` for better performance

Reply:
done